backend/svirt.pm

```diff
@@ -226,6 +249,36 @@ sub start_serial_grab {
     }
 }
 
+# opens another SSH connection to grab the serial console with the specified port
+sub open_serial_console_via_ssh {
+    my ($self, $name, $port) = @_;
+
+    # FIXME: don't hardcode path
+    my $read_serial_script = Mojo::File->new('/hdd/openqa-devel/repos/os-autoinst/read_serial_dev_from_virsh_xml.pm')->slurp;
```

Comment:
Isn't there a `virsh` command which lists the serial devices for a 'domain'?

Reply:
I looked for such a command in the first place (before writing this Perl script) but didn't find anything - except dumping the XML of course. I'm going to look again... because passing this script via SSH is really messy and so far I don't know a better way.

Reply:
This is better. However I wonder if you can change the domain config so that the serial device is exposed on the host as a socket instead of a TTY device? Then virtio/serial_screen can connect to it directly with no SSH or virsh.

I think this is probably the way VNC works, libvirt just gets the hypervisor to create a socket and the VNC console connects to it, the same as for QEMU.

Reply:
Ok, but I'd like to go for something that works and maybe improve this later. So I changed this now to use `virsh console` which already works for the serial log.

Reply:
OK, if `virsh console` passes the data through unmolested then this is probably the best solution anyway.